newrelic/resource_newrelic_alert_muting_rule.go

```diff
@@ -41,7 +43,23 @@ func resourceNewRelicAlertMutingRule() *schema.Resource {
 									"attribute": {
 										Type:         schema.TypeString,
 										Required:     true,
-										ValidateFunc: validation.StringInSlice([]string{"accountId", "conditionId", "policyId", "policyName", "conditionName", "conditionType", "conditionRunbookUrl", "product", "targetId", "targetName", "nrqlEventType", "tag", "nrqlQuery"}, false),
+										ValidateFunc:func(val interface{}, key string) (warns []string, errs []error) {
+											valueString := val.(string)
+											attemptedTagRegex := regexp.MustCompile(`^tag`)
+											correctTagRegex := regexp.MustCompile(`^tag\..+$`)
+
+											// tag.SomeValue attempted but does not match allowed format
+											if attemptedTagRegex.Match([]byte(valueString)){
+												if !correctTagRegex.Match([]byte(valueString)){
+													errs = append(errs, fmt.Errorf("%#v of %#v must be in the format tag.tag_name", key, valueString))
+													return
+												} else {
+													return
+												}
+											}
+											v := validation.StringInSlice([]string{"accountId", "conditionId", "policyId", "policyName", "conditionName", "conditionType", "conditionRunbookUrl", "product", "targetId", "targetName", "nrqlEventType", "tag", "nrqlQuery"}, false)
```

Comment:
Should this just return `validation.StringInSlice`?

Reply:
My understanding is that `validation.StringInSlice` returns a `schema.SchemaValidateFunc` which we need to call to perform the validation.

Reply:
Yeh, I see what you're saying. Returning that wouldn't return the expected type for the function, hence we return the result of calling the function.

Reply:
Yes, you're right, my misunderstanding.